theano/gpuarray/tests/test_subtensor.py

```diff
@@ -49,6 +50,46 @@ def shared(x, **kwargs):
         assert self.sub == GpuSubtensor
 
 
+def test_advinc_subtensor():
+    x_shp = (20, 15, 10, 5)
+    idx = ([[0, 1],
+            [2, 3]],
+           [[0, 1],
+            [2, 3]])
```

Comment:
Having a test case where the indices are mixing scalars and arrays would be good especially since lots of code paths are dealing with that.